src/parcel_snoopi_deep.jl

```diff
@@ -24,7 +26,45 @@ function flatten_times(timing::Core.Compiler.Timings.Timing; tmin_secs = 0.0)
         end
         append!(frontier, t.children)
     end
-    return sort(out; by=tl->tl[1])
+    return sort(out; by=first)
+end
+
+"""
+    accumulate_by_method(pairs; tmin_secs = 0.0)
+
+Add the inference timings for all `MethodInstance`s of a single `Method` together.
+`pairs` is the output of [`flatten_times`](@ref).
+Returns a list of `t => method` pairs.
+
+When the accumulated time for a `Method` is large, but each instance is small, it indicates
+that it is being inferred for many specializations (which might include specializations for different constants).
+
+# Example
+
+```julia
+julia> tinf = @snoopi_deep sort(Float16[1, 2, 3]);
+
+julia> tm = accumulate_by_method(flatten_times(tinf); tmin_secs=0.0005)
+6-element Vector{Pair{Float64, Method}}:
+ 0.000590579 => _copyto_impl!(dest::Array, doffs::Integer, src::Array, soffs::Integer, n::Integer) in Base at array.jl:307
+ 0.000616788 => partition!(v::AbstractVector{T} where T, lo::Integer, hi::Integer, o::Base.Order.Ordering) in Base.Sort at sort.jl:578
+ 0.000634394 => sort!(v::AbstractVector{T} where T, lo::Integer, hi::Integer, ::Base.Sort.InsertionSortAlg, o::Base.Order.Ordering) in Base.Sort at sort.jl:527
+ 0.000720815 => Vector{T}(::UndefInitializer, m::Int64) where T in Core at boot.jl:448
+ 0.001157551 => getindex(::Type{T}, x, y, z) where T in Base at array.jl:394
+ 0.046509861 => ROOT() in Core.Compiler.Timings at compiler/typeinfer.jl:75
+```
+
+`ROOT` is a dummy element whose time corresponds to the total time for the operation, not just inference time.
+"""
+function accumulate_by_method(pairs::Vector{Pair{Float64,Core.Compiler.Timings.InferenceFrameInfo}}; tmin_secs = 0.0)
+    tmp = Dict{Method,Float64}()
+    for (t, info) in pairs
+        m = info.mi.def
+        if isa(m, Method)
+            tmp[m] = get(tmp, m, 0.0) + t
+        end
+    end
+    return sort([t=>m for (m, t) in tmp if t >= tmin_secs]; by=first)
 end
```

Comment:
This is perfect! 👌!

I should've realized we needed this function a while ago -- I've been just reusing this line from my julia REPL history 😬 !:
```julia
julia> ms = let ms = DefaultDict{Any,Float64}(0)
           for (t,mi_info) in times
               ms[mi_info.mi.def] += t
           end
           ms
       end;

julia> methods_sorted = sort([v=>k for (k,v) in ms], by=first)
```

It looks exactly identical to what you've done here 👍 👌

Reply:
One slight difference is that you allow `m` to be a `Module`. If we include modules then we might need to reconsider the name.

Reply:
oh good point. I hadn't realized I think that `m` could be a Module...

I'm not sure what's best. I guess if one of the MethodInstances where `m` is a `Module` _was_ accounting for most of the time, I'd be sad if we missed it because we were filtering those out! So maybe we should include them too?

I dunno if we really need to change the name, since it's _usually_ a method? Or maybe we could call it `accumulate_by_definition()`, to be more inclusive?

Reply:
OK, now the logic is:
- if `mi.def` is a Method, aggegation makes sense and use the Method as the key of the dict
- if `mi.def` is a module, it's perfectly possible for there to be multiple thunks from the same module. Hence, we just preserve `mi`.